src/transformations/correction/intersecting_polygons.jl

```diff
@@ -59,13 +59,78 @@ struct UnionIntersectingPolygons <: GeometryCorrection end
 application_level(::UnionIntersectingPolygons) = GI.MultiPolygonTrait
 
 function (::UnionIntersectingPolygons)(::GI.MultiPolygonTrait, multipoly)
-    union_multipoly = if GI.npolygon(multipoly) > 1
+    union_multipoly = tuples(multipoly)
```

Comment:
Amazing @rafaqz ! Yeah, I was wondering if it was that easy to get the extent, so that is great to know. However, what do you think about it being a user input? I think we might need to dispatch off of the Bools as Types stuff @asinghvi17 implemented. With `calc_extent = true` we have the following:
```julia
mp6 = GI.MultiPolygon([[[(-5.0, 10.0), (-5.0, 15.0), (15.0, 15.0), (15.0, 10.0), (-5.0, 10.0)]], [[(-5.0, -5.0), (-5.0, 0.0), (15.0, 0.0), (15.0, -5.0), (-5.0, -5.0)]], [[(10.0, -5.0), (10.0, 15.0), (15.0, 15.0), (15.0, -5.0), (10.0, -5.0)]], [[(-5.0, -5.0), (-5.0, 15.0), (0.0, 15.0), (0.0, -5.0), (-5.0, -5.0)]]])
typeof(GO.tuples(mp6; calc_extent = true))
```
which returns 
`GeoInterface.Wrappers.MultiPolygon{false, false, Vector{GeoInterface.Wrappers.Polygon{false, false, Vector{GeoInterface.Wrappers.LinearRing{false, false, Vector{Tuple{Float64, Float64}}, Extents.Extent{(:X, :Y), Tuple{Tuple{Float64, Float64}, Tuple{Float64, Float64}}}, Nothing}}, Extents.Extent{(:X, :Y), Tuple{Tuple{Float64, Float64}, Tuple{Float64, Float64}}}, Nothing}}, Extents.Extent{(:X, :Y), Tuple{Tuple{Float64, Float64}, Tuple{Float64, Float64}}}, Nothing}
`

while
```julia
typeof(GO.tuples(mp6; calc_extent = false))
```
returns
`GeoInterface.Wrappers.MultiPolygon{false, false, Vector{GeoInterface.Wrappers.Polygon{false, false, Vector{GeoInterface.Wrappers.LinearRing{false, false, Vector{Tuple{Float64, Float64}}, Nothing, Nothing}}, Nothing, Nothing}}, Nothing, Nothing}`

Seems like that will cause type instability if we don't dispatch off of user input. 

And if that is the case, I would rather have that be its own PR where I add extents to all of the polygon functions I. have written because I don't think I took advantage of it as much as I should have.

Reply:
This is a part of the problems listed in #97

Reply:
Yeah use BoolAsTypes! its just easier to use `true` for demonstration purposes

Reply:
Are you okay with this being in a different PR? If so, I might just merge and then work on that when I have a bit more time (and once I am done with #106 😆).

Reply:
For sure :)

Otherwuse its all good so I jumped ahead to optimizing

Reply:
ok maybe I can PR with the extent stuff too if you have other priorities, seeing its mostly building on my code in other packages. Unfortunately I've been stuck with a lot of DD/Rasters work the last month and havent had time for looking at clipping performance here yet

Reply:
I am also happy to take a stab at it. It just isn't a priority like getting these bugs fixed or getting the points swapped over for me right now. But I am excited for one of us to work on it in the near future haha